### PR TITLE
[MINDEXER-145] Update parent and dependencies

### PR DESCRIPTION
**Pull request by cstamas**
Updates parent POM to 35 and several dependencies (minor).

---

https://issues.apache.org/jira/browse/MINDEXER-145

### CODE REVIEW

**Review by slawekjaranowski — CHANGES_REQUESTED** [review]
please also override 
 - m-pmd-p to 3.16.0
 - taglist to 3.0.0
 
please look at generated site reports

**Review by slachiewicz — APPROVED** [review]

File: indexer-examples/indexer-examples-spring/pom.xml [hunk 1]
```diff
@@ -35,7 +35,7 @@ under the License.
   </description>
 
   <properties>
-    <version.spring>5.3.15</version.spring>
+    <version.spring>5.3.16</version.spring>
```
Comment (michael-o) [inline indexer-examples/indexer-examples-spring/pom.xml:38]: This needs already an update by now.

**Comment by jira-importer** [comment]
Resolve #681

**Comment by jira-importer** [comment]
Resolve #681

**Pull request closed without merging**